src/EurekaClient.js

```diff
@@ -426,10 +430,22 @@ export default class Eureka extends EventEmitter {
   eurekaRequest(opts, callback, retryAttempt = 0) {
     this.clusterResolver.resolveEurekaUrl((err, eurekaUrl) => {
       if (err) return callback(err);
-      const requestOpts = merge({
+      let requestOpts = merge({
         baseUrl: eurekaUrl,
         gzip: true,
       }, opts);
+
+      if (this.requestMiddleware) {
+        try {
+          requestOpts = this.requestMiddleware(requestOpts);
```

Comment:
Awesome! Yeah, we were intending on plugging in logic to get/acquire/refresh an access token depending on whether or not you had one and whether it was expired yet. Thanks!